MCprep_addon/spawner/models.py

```diff
@@ -0,0 +1,1350 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+
+#import math
+import os
+#import random
+#import time
+
+import bpy
+#import mathutils
+
+# addon imports
+from .. import conf
+from ..materials import generate
+from .. import util
+from .. import tracking
+
+
+# -----------------------------------------------------------------------------
+# Mesh swap functions
+# -----------------------------------------------------------------------------
+
+model_cache = {}
+model_cache_path = None
+
+def get_model_cache(context, clear=False):
```

Comment:
FYI only, no action required for you: I am going to think how to refactor this so it's also in a way configurable and shared code with meshswap, noting it would have to be further abstracted since it specifically references the according entity or meshswap properties.